### PR TITLE
[WIP] DOC/MNT: Refactor datasets

### PR DESCRIPTION
**Pull request by massich**

### CODE REVIEW

**Comment by larsoner** [comment]
What's the advantage of this?

File: doc/manual/datasets_index.rst [hunk 1]
```diff
@@ -1,262 +0,0 @@
...
-    from mne.datasets import eegbci
-    raw_fnames = eegbci.load_data(subject, runs)
-    raws = [read_raw_edf(f, preload=True) for f in raw_fnames]
-    raw = concatenate_raws(raws)
-
-.. topic:: Examples
-
-    * :ref:`sphx_glr_auto_examples_decoding_plot_decoding_csp_eeg.py`
-
-Do not hesitate to contact MNE-Python developers on the
```
Comment (massich) [inline doc/manual/datasets_index.rst]: This is general to datasets not to `EEGBCI motor imagery`

**Comment by massich** [comment]
I agree that if we had fewer datasets and each of them had longer descriptions it would then it would be necessary (which now is not necessary).

But I think that by restructuring it like this, would avoid errors like [this](https://github.com/mne-tools/mne-python/pull/5720/files#r234291646).

More over it would allow to more detailed description of the datasets, and I find easier to tell someone to add `manual/datasets/awesome_dataset.rst` and link it in the index.

but I agree that is just a personal taste

**Comment by massich** [comment]
and I might also be a YAGNI, until we want to group the datasets somwehow with some narrative. I would also move the data descriptions in the tutorials to here. And link from there.

**Comment by massich** [comment]
And maybe the sample_dataset.rst should actually be the content of `datasets/sample.rst`

**Comment by agramfort** [comment]
I am +0 on this. I don't see a clear benefit for doing this also. But whatever.

**Comment by sappelhoff** [comment]
I actually like this, but as @massich mentioned, it might be more personal taste.

**Pull request closed without merging**